File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
	public static String solution(String[] participant, String[] completion) {
		HashMap<String, Integer> partiMap = new HashMap<>();

		int temp = 0;

		for (String s : completion) {
			if (partiMap.containsKey(s)) {
				temp = partiMap.get(s);
				partiMap.put(s, temp + 1);
			}
			else 
				partiMap.put(s, 1);
		}

		for (String s : participant) {
			if (partiMap.containsKey(s)) {
				temp = partiMap.get(s);
				partiMap.put(s, temp - 1);
			}
			if (partiMap.get(s) == null || partiMap.get(s) < 0) //듣보잡 이름 || 동명이인 처리
				return s;
		}

		String answer = "";
		return answer;
	}
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map <string, int> mapNames;

    for (string name : participant)
    {
        ++mapNames[name];
    }

    for (string delName : completion)
    {
        --mapNames[delName];
    }

    for (auto iter : mapNames)
    {
        if (1 == iter.second)
            return iter.first;
    }

    return "";
}